soroban-env-host/src/host.rs

```diff
@@ -1074,6 +1074,31 @@ impl VmCallerCheckedEnv for Host {
         Ok(RawVal::from_void())
     }
 
+    fn log_fmt_values(
+        &self,
+        _vmcaller: &mut VmCaller<Host>,
+        fmt: Object,
+        args: Object,
+    ) -> Result<RawVal, HostError> {
+        if cfg!(debug_assertions) {
+            let fmt: String = self
+                .visit_obj(fmt, move |hv: &Vec<u8>| Ok(String::from_utf8(hv.clone())))?
+                // TODO: Remove unwrap.
+                .unwrap();
+            let args: HostVec = self.visit_obj(args, move |hv: &HostVec| Ok(hv.clone()))?;
+            self.record_debug_event(
+                DebugEvent::new()
+                    .msg(fmt)
+                    .args(args.iter().map(|arg| arg.to_raw())),
+            )?;
+            Ok(RawVal::from_void())
+        } else {
+            Err(self.err_general(
+                "log_fmt_values unsupported outside of debug-assertion builds of the host",
+            ))
```

Comment:
We actively error if someone tries to execute a debug build on a host that does not have debug enabled. I think this is preferred over doing nothing as this ensures we will never have contracts deployed and in use that log. The downside of this is if a developer accidentally releases then finds out sometime later they left debug on and one of their functions they need fails.

I don't mind what we do here, maybe it would be saferto leave this error out and make it a no-op.

Wdyt @graydon?

Reply:
I changed it to a no-op out of concern it would create less problems for contracts that are accidentally deployed with a debug log.